pkg/queues/redis_provider.go

```diff
@@ -986,12 +1032,23 @@ func (r *redisQueue) ProcessTimedOutMessages(ctx context.Context, timeout time.D
 	failedKey := fmt.Sprintf("failed_messages:%s", r.name)
 	timedOutCount := 0
 
+	if err := r.ensureConsumerGroup(ctx); err != nil {
+		return 0, err
+	}
+
 	// Get pending messages using XPENDING
 	pending, err := r.client.XPending(ctx, r.name, r.groupName).Result()
 	if err != nil {
-		return 0, fmt.Errorf("failed to get pending messages: %w", err)
+		if strings.Contains(err.Error(), "NOGROUP") {
+			if ensureErr := r.ensureConsumerGroup(ctx); ensureErr != nil {
+				return 0, ensureErr
+			}
+			pending, err = r.client.XPending(ctx, r.name, r.groupName).Result()
+		}
+		if err != nil {
```

Comment:
nit: you have scope issue here the err at line 1040 can be replaced by an error on 1046 please revisit